src/module_2/module_2_part_1.md

```diff
@@ -0,0 +1,1349 @@
+      <td>33</td>
+      <td>573.32</td>
+      <td>884.29</td>
+    </tr>
+    <tr>
+      <th>40</th>
+      <td>premixed-cocktails</td>
+      <td>29</td>
+      <td>9</td>
+      <td>556.21</td>
+      <td>725.62</td>
+    </tr>
+    <tr>
+      <th>41</th>
+      <td>sexual-health</td>
+      <td>42</td>
+      <td>12</td>
+      <td>538.08</td>
+      <td>703.96</td>
+    </tr>
+    <tr>
+      <th>33</th>
+      <td>medicines-treatments</td>
+      <td>97</td>
+      <td>72</td>
+      <td>384.63</td>
+      <td>663.31</td>
+    </tr>
+    <tr>
+      <th>30</th>
+      <td>low-no-alcohol</td>
+      <td>58</td>
+      <td>39</td>
+      <td>294.46</td>
+      <td>527.55</td>
+    </tr>
+    <tr>
+      <th>31</th>
+      <td>maternity</td>
+      <td>40</td>
+      <td>20</td>
+      <td>283.40</td>
+      <td>424.10</td>
+    </tr>
+    <tr>
+      <th>0</th>
+      <td>adult-incontinence</td>
+      <td>39</td>
+      <td>7</td>
+      <td>278.61</td>
+      <td>357.14</td>
+    </tr>
+    <tr>
+      <th>48</th>
+      <td>suncare</td>
+      <td>38</td>
+      <td>13</td>
+      <td>271.12</td>
+      <td>360.49</td>
+    </tr>
+    <tr>
+      <th>32</th>
+      <td>medicine-treatments</td>
+      <td>27</td>
+      <td>30</td>
+      <td>103.83</td>
+      <td>216.23</td>
+    </tr>
+    <tr>
+      <th>34</th>
+      <td>mixed-bundles</td>
+      <td>2</td>
+      <td>3</td>
+      <td>64.98</td>
+      <td>207.95</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+We can see the net revenue per product type, which is caluclated using the total sales. There's a huge difference between some of the product types; which basically can lead to either focusing on increasing those sales of the products with less revenue, maybe with more ads or any marketing strategy for it, or just focusing in a very specific nicht. Also we can see that due to abandoned carts there's being lost a lot of possible income to the bussiness. Therefore, another important aspect we could analyse is how to reduce the amount of these abandoned carts. When are more succesful orders? Should we create a strategy to increase sales at determined times? 
+
+
+```python
+def best_users_product_type():
+    # Mapear variant_id a product_type
+    variant_info = df_inventory.set_index('variant_id')[['product_type', 'price']].to_dict('index')
+
+    # Expandir los items ordenados y abandonados en df_orders y df_abandoned_carts
+    expanded_orders = df_orders.explode('ordered_items')
+    expanded_abandoned = df_abandoned_carts.explode('variant_id')
+
+    # Filtrar variant_id en df_orders y df_abandoned_carts que están en df_inventory
+    valid_variants = set(df_inventory['variant_id'])
+    expanded_orders = expanded_orders[expanded_orders['ordered_items'].isin(valid_variants)]
+    expanded_abandoned = expanded_abandoned[expanded_abandoned['variant_id'].isin(valid_variants)]
+
+    # Asignar product_type y price a cada variant_id en los pedidos y abandonos
```

Comment:
This is ineficient and makes the code less clear. Why don't you use `pd.merge` map and apply? 

In this case, all you need is to explode products where it is a list, and do a `inner join` with `inventory_df` that way you will be filtering product not available in `inventory_df` and adding any relevant column from it.